__tests__/footprint.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { focusMerge } from '../src';
+
+describe('Complexity footprint', () => {
+  const createTest = () => {
+    document.body.innerHTML = `    
+    <div id="d1"> 
+       <button id="b1">1</button>
+       <button>2</button>       
+    </div>    
+    <div id="d2"> 
+       <button id="b2">1</button>
+       <button>2</button>
+    </div>
+    `;
+  };
+
+  const querySelector = (q: string): HTMLElement => document.querySelector<HTMLElement>(q)!;
+
+  beforeEach(() => {
+    const { getComputedStyle } = window;
+    jest.spyOn(window, 'getComputedStyle').mockImplementation(getComputedStyle);
+  });
+
+  afterEach(() => {
+    ((window.getComputedStyle as unknown) as jest.SpyInstance).mockRestore();
+  });
+
+  it('known operation complexity - no focus', () => {
+    createTest();
+    focusMerge(querySelector('#d1'), null);
+    expect(window.getComputedStyle).toBeCalledTimes(32);
+  });
+
+  it('known operation complexity - has focus inside', () => {
+    createTest();
+    querySelector('#b1').focus();
+    focusMerge(querySelector('#d1'), null);
+    expect(window.getComputedStyle).toBeCalledTimes(24);
```

Comment:
- d1: - 2 buttons + d1 + body === 24 operations
- 3 buttons -> 36 operations, 

which lead to 12 operations PER one target focusable

d2 adds 8 operations for 2 buttons by hoisting "common parent"

Total times the "check" is called - 3 times (2 for inner nodes and one for all)